packages/flutter/lib/src/rendering/object.dart

```diff
@@ -978,19 +1005,42 @@ class PipelineOwner {
         arguments: debugTimelineArguments,
       );
     }
-    assert(() {
-      _debugDoingPaint = true;
-      return true;
-    }());
     try {
-      final List<RenderObject> dirtyNodes = _nodesNeedingPaint;
+      List<RenderObject> dirtyNodes = _nodesNeedingPaint;
+      _nodesNeedingPaint = <RenderObject>[];
+      // It is possible that one or more render objects in the paint list are no
+      // longer repaint boundaries. Instead of painting from these nodes, the
+      // nearest parent render object repaint boundary should be used instead.
+      bool updated = false;
+      for (final RenderObject object in dirtyNodes) {
```

Comment:
This seems odd. Don't we get notified when you cease to be a repaint boundary? Why can we not do this work then just for the affected node rather than iterating over everything here?

Reply:
ahh yeah, you'd call markNeedsCompositingBitsUpdate. That is called before flush paint so we should handle that there instead